prov/usnic/src/usdf_dgram.c

```diff
@@ -131,30 +131,32 @@ usdf_dgram_recvmsg(struct fid_ep *fep, const struct fi_msg *msg, uint64_t flags)
 		msg->iov_count, (fi_addr_t)msg->addr, msg->context);
 }
 
-static inline ssize_t
-_usdf_dgram_send(struct usdf_ep *ep, struct usdf_dest *dest,
-		const void *buf, size_t len,  void *context)
-{
-	if (len <= USD_SEND_MAX_COPY - sizeof(struct usd_udp_hdr)) {
-		return usd_post_send_one_copy(ep->e.dg.ep_qp,
-			&dest->ds_dest, buf, len, USD_SF_SIGNAL, context);
-	} else {
-		return usd_post_send_one(ep->e.dg.ep_qp, &dest->ds_dest,
-			buf, len, USD_SF_SIGNAL, context);
-	}
-}
-
 ssize_t
 usdf_dgram_send(struct fid_ep *fep, const void *buf, size_t len, void *desc,
 		fi_addr_t dest_addr, void *context)
 {
 	struct usdf_ep *ep;
+	uint8_t completion;
 	struct usdf_dest *dest;
 
 	ep = ep_ftou(fep);
-
 	dest = (struct usdf_dest *)(uintptr_t) dest_addr;
-	return _usdf_dgram_send(ep, dest, buf, len, context);
+
+	if (ep->ep_tx_dflt_signal_comp ||
+		ep->e.dg.tx_op_flags & FI_COMPLETION)
+		completion = USD_SF_SIGNAL;
```

Comment:
Good catch. This one is unique in that it's the only one that needs either an else clause or an initial value. Seem to have missed that.